src/core/login/login-stash.ts

```diff
@@ -75,20 +75,42 @@ export interface LoginStash {
 
 /**
  * Reads all login stashes from disk.
+ *
+ * It will apply any temporary files as overrides. A temporary file is an
+ * override file that ends with `-temp.json`. See {@link saveStashTemporarily}
+ * and {@link removeStashTemporarily} as usage examples.
  */
 export async function loadStashes(
   disklet: Disklet,
   log: EdgeLog
 ): Promise<LoginStash[]> {
   const out: LoginStash[] = []
+  const temporaryOverrides: LoginStash[] = []
   const paths = await disklet.list('logins').then(justFiles)
   for (const path of paths) {
     try {
-      out.push(asLoginStash(JSON.parse(await disklet.getText(path))))
+      const stash = asLoginStash(JSON.parse(await disklet.getText(path)))
+      if (path.endsWith('-temp.json')) {
+        temporaryOverrides.push(stash)
+        continue
+      }
```

Comment:
We don't want to add a whole new file, because disk is slow. Instead, we want to add fields to the `LoginStash` type. Something like:

```ts
interface LoginStash {
  ...
  // Tell the truth if these are undefined:
  pinEnabledLie?: boolean
  duressEnabledLie?: boolean
  ...
}
```

And then where we would normally do `pinEnabled = <existing-logic>`, we just do `pinEnabled = pinEnabledLie ?? <existing-logic>`. Pretty straightforward.

Reply:
What you propose will work, but would this mean that "online logins" will stomp on these fields? If so, then my [next commit](72d334d13136412ed0dc4f7e54657ac4a0577b68) which changes the spec is not possible, which means that the [first commit](fcd0aa5189888947aa38c0489caa2a53c98aa597) must be accepted in full.

I can try you're suggestion here and if it doesn't stomp, then no further discussion needed.

Reply:
Server logins never stomp the stash. That's why we use `updateTree` and require an existing stash - we can always preserve the existing fields by copying them over as part of the update.